packages/mobile/src/screens/collection-screen/useCollectionLineup.ts

```diff
@@ -37,43 +38,60 @@ export const useCollectionLineup = (
 ) => {
   const dispatch = useDispatch()
   const isOfflineModeEnabled = useIsOfflineModeEnabled()
-  const offlineTracks = useSelector(getOfflineTracks)
+  const offlineTrackIds = useSelector(
+    (state) => new Set(getOfflineTrackIds(state) || []),
+    areSetsEqual
+  )
   const isReachable = useSelector(getIsReachable)
   const collection = useSelector((state) => {
     return getCollection(state, { id: collectionId as number })
   })
   const collectionTracks = useSelector(getCollectionTracksLineup)
   const collectionTrackUidMap = collectionTracks.entries.reduce(
     (acc, track) => {
-      acc[track.id] = track.uid
+      if (acc[track.id] && acc[track.id].includes(track.id)) {
+        return acc
+      }
+      acc[track.id] = acc[track.id]
+        ? acc[track.id].concat(track.uid)
+        : [track.uid]
```

Comment:
it feels like this could be combined into the map below, but I think it works as is